optim/aref.lisp

```diff
@@ -50,23 +50,10 @@
                                         `(%dense-array ,elt-type ,rank)))
              (subscript-types (mapcar (lm form (primary-form-type form env)) subscripts))
              (ds              (make-gensym-list (length subscripts) "DIMENSION"))
-             (os              (make-gensym-list (length subscripts) "OFFSET"))
              (ss              (make-gensym-list (length subscripts) "STRIDE"))
              (optim-expansion
                (once-only (array)
                  (cond
-                   ((eql rank 0)
```

Comment:
I'm possibly missing something. Is there a reason to delete the rank 0 cases for the compiler macro?

Reply:
The optimization used to assume offset=0 for rank 0 array (which was true because rank 0 array didn't properly support offset in the first place). Now this is no longer the case and rank 0 array should be treated the same as the general case.

Reply:
Ah, right, that makes sense! Thanks!